zip-proportional-output-fee-mechanism-pofm.rst

```diff
@@ -0,0 +1,191 @@
+::
+
+  ZIP: XXX
+  Title: Proportional Output Fee Mechanism (POFM)
+  Owners: Aditya Bharadwaj <nighthawk24@gmail.com>
+  Credits: Madars Vizra, Kris Nuttycombe
+  Status: Draft
+  Community Status: Request for comments : 
+  Category: Wallet
+  Created: 2022-08-15
+  License: MIT
+
+
+Terminology
+===========
+
+The key words "MUST", "SHOULD", "SHOULD NOT", "MAY", "RECOMMENDED",
+"OPTIONAL", and "REQUIRED" in this document are to be interpreted as
+described in RFC 2119. [#RFC2119]_
+
+"Z" refers to shielded address.
+"T" refers to transparent address.
+
+"POFM" refers to the proposed change in conventional fee as described in 
+this ZIP.
+
+The term "default transaction fee" in this document is in reference
+to Z to Z, T to Z & Z to T transaction fee paid to miners on the Zcash network
+for the work on including the shielded transaction in a block.
+
+
+Abstract
+========
+
+The goal of this ZIP is to change the conventional transaction fees for
+Shielded transactions and get buy-in from wallet developers, miners & Zcash users.
+
+With an updated transaction fee formula, miners will be compensated fairly for
+including transactions with a high number of outputs, while still allowing low fees 
+for regular shielded transaction use cases.
+
+
+Out of Scope for this Proposal
+============
+
+Discussion around transaction fees for T to T.
+
+
+Motivation
+============
+
+In light of recent network activity, it is time to review and update the 
+standard 1,000 zats transaction fees set in ZIP-313.
+
+
+The conventional transaction fee presently is 0.00001 ZEC or 1,000 zats per
+ZIP-313, that allowed exploration of novel use cases of the Zcash blockchain.
+The Zcash network has operated for almost 2 years of 1,000 zats fee per shielded 
+transaction without consideration for the total number of inputs and outputs in the transaction.
+This has resulted in high output transactions with 1,100 outputs costing the same as 
+transactions with 2 outputs.
+
+
+Requirements for consensus
+-------------------------
+
+Wallet developers must update the fees to the proposed formula by Madars and 
+Kris Nuttycomb [#madars-1]_
+
+min_fee = 0.00001 + 0.00001 * max(0, #nullifiers + #shielded outputs - 4)
```

Comment:
I believe that it would be better to charge:
```
min_fee = base_fee * max(0, #inputs + #outputs - 4)`
```
where `#inputs` and `#outputs` also take into account transparent inputs and outputs. Otherwise, the fee structure (if not otherwise changed) will preferentially encourage usage of the transparent part of the chain.

Something that the community might wish also to consider is imposing a surcharge for transparent inputs and outputs to encourage the use of the shielded pool, perhaps by using a different `base_fee` value for transparent inputs and outputs.

Reply:
Agree, that this might be a good time to include a higher `base_fee` for transparent transactions.

Hey, @daira would you support the idea of making a change to both transparent and shielded fee costs in this ZIP?

Reply:
Does `min_fee = base_fee * max(0, #inputs + #outputs - 4)` allow for a min_fee of 0? Should that change to `max(1, ...)`?

Reply:
Oops, yes, that was my error.

Reply:
Thank you @AloeareV, I've updated the formula.

Reply:
@daira @nuttycom How about setting the `base_fee` of 0.0001 or 10,000 zatoshi for Transparent ZEC & 0.00001 or 1,000 zatoshi for Shielded, Shielding & De-shielding transactions?

Reply:
I don't have an opinion on this, those kinds of choices should be made with broader community input.

Reply:
One concern that should be taken into account is the fee behavior of transparent-only wallets. Most of them copy their fee logic from Bitcoin fee code, which generally charges much higher fees than what is being discussed here, but it's important that such wallets (that don't usually respond quickly to zcash ecosystem changes) continue to function.